src/matrix.c

```diff
@@ -0,0 +1,72 @@
+#include "matrix.h"
+
+#include <math.h>
+
+inline matrix_size coordToIndex(const matrix_size row, const matrix_size col, const matrix_size size) {
+    return row * size + col;
+}
+
+inline matrix_size isDiagonal(const matrix_size index, const matrix_size size) {
+    return ((index / size) == (index % size));
+}
+
+void matrixMultiply(const matrix_real_number* firstMatrix, const matrix_real_number* secondMatrix, const matrix_size size[3], matrix_real_number* output, const matrix_size transposeFirstMatrix) {
+    const matrix_size m = size[0];
+    const matrix_size n = size[1];
+    const matrix_size p = size[2];
+    for (matrix_size i = 0; i < m; ++i) {
+        for (matrix_size j = 0; j < p; ++j) {
+            matrix_real_number sum = 0.0;
+            for (matrix_size k = 0; k < n; ++k) {
+                sum += (transposeFirstMatrix ? firstMatrix[coordToIndex(k, i, n)] : firstMatrix[coordToIndex(i, k, n)]) * secondMatrix[coordToIndex(k, j, p)];
+            }
+            output[coordToIndex(i, j, p)] = sum;
+        }
+    }
+}
+
+void createIdentityMatrix(const matrix_size size, matrix_real_number* output) {
+    const matrix_size sizeSquared = size * size;
+    for (matrix_size i = 0; i < sizeSquared; ++i) {
+        if (isDiagonal(i, size)) {
+            output[i] = 1;
+        } else {
+            output[i] = 0;
+        }
+    }
+}
+
+void transpose(matrix_real_number* input, matrix_real_number* output, matrix_size* dims) {
```

Comment:
I would name that function `matrixTranspose` to respect the convention defined by the other functions or I would rename other functions to respect that convention (for example simply put `multiply` for the multiplication function